packages/core/src/components/Tooltip/index.tsx

```diff
@@ -213,8 +226,52 @@ export class Tooltip extends React.Component<TooltipProps & WithStylesProps, Too
 
   private handleMouseLeave = () => {
     if (!this.props.toggleOnClick) {
-      this.handleClose();
+      if (this.props.popover) {
+        this.delaySetPopupVisible(false, MOUSE_LEAVE_DELAY);
+      } else {
+        this.handleClose();
+      }
+    }
+  };
+
+  setPopupVisible(open: boolean) {
+    const { open: previousOpen } = this.state;
+    this.clearDelayTimer();
+    if (previousOpen !== open) {
+      if (open) {
+        this.handleOpen();
+      } else {
+        this.handleClose();
+      }
+    }
+  }
+
+  clearDelayTimer() {
+    if (this.delayTimer) {
+      clearTimeout(this.delayTimer);
+      this.delayTimer = null;
+    }
+  }
+
+  delaySetPopupVisible(open: boolean, delayS: number) {
```

Comment:
this name is a little confusing to me. what do you think about `delayedSetPopoverVisible`?  (also `popup` => `popover` to be consistent?)